### PR TITLE
Benchmarking

### PR DESCRIPTION
**Pull request by ffreyer**
I added a macro `@bm` which wraps the body of a function in `@timeit_debug <name of function> begin ... end`. Imo this is nicer than writing `@timeit_debug` in every function. It's also easier/faster to do.

I think it would be fine to tag functions with `@bm` in master, since `@timeit_debug` should be optimized away when benchmarks are disabled and because it is relatively little noise. Thoughts?

### CODE REVIEW

**Comment by ffreyer** [comment]
Iirc Carsten was in favor of using `@bm` on master.

The last commit adds `@bm` to 
- `run!`, `sweep_spatial` (DQMC.jl)
- `run!`, `sweep` (MC.jl)
- `add_slice_sequence_left/right`, `calculate_greens`, `calculate_logdet`, `wrap_greens!`, `propagate` (stack.jl)
- `measure!` (flavors/DQMC/measurements.jl, Measurements.jl)
- `propose_local`, `accept_local`, `interaction_matrix_exp` (HubbardModelAttractive.jl)
- `propose_local`, `accept_local`, `global_move` (IsingModel.jl)

I think this can be merged now.

**Comment by ffreyer** [comment]
This closes #33

File: src/MonteCarlo.jl [hunk 1]
```diff
@@ -1,19 +1,19 @@
 module MonteCarlo
 
 using Reexport
-@reexport using MonteCarloObservable
+@reexport using MonteCarloObservable, TimerOutputs
```
Comment (carstenbauer) [inline src/MonteCarlo.jl]: Why reexport TimerOutputs? I'd prefer to not do this.

**Comment by carstenbauer** [comment]
Apart from the `@reexport TimerOutputs` we can merge this.

File: Project.toml [hunk 2]
```diff
@@ -17,6 +17,7 @@ Reexport = "189a3867-3050-52da-a836-e630ba90ab69"
 Requires = "ae029012-a4dd-5104-9daa-d747884805df"
 SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 StableDQMC = "0827807b-44c9-51d0-af97-efe667fe6b2a"
+TimerOutputs = "a759f4b9-e2f1-59dc-863e-4aeb61b1ea8f"
```
Comment (carstenbauer) [inline Project.toml:20]: Needs a compat entry.

**Review by carstenbauer — CHANGES_REQUESTED** [review]

File: src/MonteCarlo.jl (hunk 1 above)
Comment (ffreyer) [reply src/MonteCarlo.jl]: I wanted `print_timer()`, `reset_timer!()` and `TimerOutputs.enable_debug_timings(module)` to be available. Guess we can also hide them...

Comment (carstenbauer) [reply src/MonteCarlo.jl]: I think it's fine to have to load TimerOutputs manually for printing the timings. Perhaps a function `enable_benchmarks` that wraps `TimerOutputs.enable_debug_timings(MonteCarlo)` would be handy.

**Comment by ffreyer** [comment]
Should be ready for merge now

**Pull request merged**